pkg/virt-controller/watch/vm.go

```diff
@@ -2713,18 +2714,24 @@ func (c *VMController) handleDynamicInterfaceRequests(vm *virtv1.VirtualMachine,
 		return nil
 	}
 
-	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
+	vmTemplateCopy := vm.Spec.Template.DeepCopy()
 	for i := range vm.Status.InterfaceRequests {
-		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
+		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMTemplate(
 			vmTemplateCopy, &vm.Status.InterfaceRequests[i])
 	}
-	vm.Spec.Template.Spec = *vmTemplateCopy
+	vm.Spec.Template = vmTemplateCopy
 
 	if vmi != nil && vmi.DeletionTimestamp == nil {
+		updatedVMIfaces := vmispec.IndexInterfaceSpecByName(vm.Spec.Template.Spec.Domain.Devices.Interfaces)
 		vmiSpecCopy := vmi.Spec.DeepCopy()
 		for i := range vm.Status.InterfaceRequests {
+			var macAddress string
+			if vm.Status.InterfaceRequests[i].AddInterfaceOptions != nil {
+				ifaceName := vm.Status.InterfaceRequests[i].AddInterfaceOptions.Name
+				macAddress = updatedVMIfaces[ifaceName].MacAddress
+			}
 			vmiSpecCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
-				vmiSpecCopy, &vm.Status.InterfaceRequests[i])
+				vmiSpecCopy, &vm.Status.InterfaceRequests[i], macAddress)
```

Comment:
The mac address is needed only for the addition, therefore, it would be cleaner to add the argument to that  operation and not to both addition and removal.

If I continue the split done in #9979, then it will look like this: 
```go
switch {
case request.AddInterfaceOptions != nil:
	vmiSpecCopy = controller.ApplyNetworkInterfaceAddRequest(vmiSpecCopy, request.AddInterfaceOptions, macAddress)
case request.RemoveInterfaceOptions != nil:
	vmiSpecCopy = controller.ApplyNetworkInterfaceRemoveRequest(vmiSpecCopy, request.RemoveInterfaceOptions)
}
```

Another option is to for-loop again and do the same on the relevant interfaces as an additional step.

Reply:
We can have this change once this PR get in and 9979 will rebase on it, or the other way around, whoever get merged first.

Reply:
It's not possible to do the MAC assignment as part of  `ApplyNetworkInterfaceAddRequest()` because it is also used by `ApplyNetworkInterfaceRequestOnVMISpec() which in another thread it was asked to reuse the function.